### PR TITLE
ux friendly use of remixd inside electron

### PR DESCRIPTION
**Pull request by yann300**
merge after https://github.com/ethereum/remix-ide/pull/2348

### CODE REVIEW

File: src/app/files/remixd-handle.js [hunk 1]
```diff
@@ -55,27 +56,31 @@ export class RemixdHandle extends Plugin {
     * @param {String} txHash    - hash of the transaction
     */
   connectToLocalhost () {
+    let connection = (error) => {
+      if (error) {
+        console.log(error)
+        modalDialogCustom.alert(
+          'Cannot connect to the remixd daemon.' +
+          'Please make sure you have the remixd running in the background.'
```
Comment (GrandSchtroumpf) [inline src/app/files/remixd-handle.js]: Is it run by default ? Maybe could be good to have a link to documentation.

File: src/app/files/remixd-handle.js [hunk 2]
```diff
@@ -85,6 +90,10 @@ export class RemixdHandle extends Plugin {
           }
         }
       )
+    } else {
+      this.locahostProvider.init((error) => {
```
Comment (GrandSchtroumpf) [inline src/app/files/remixd-handle.js]: Could be on one line : `this.locahostProvider.init(error => connection(error))`

File: src/remixAppManager.js [hunk 3]
```diff
@@ -19,11 +19,14 @@ export class RemixAppManager extends PluginEngine {
   constructor (plugins) {
     super(plugins, settings)
     this.event = new EventEmitter()
+    this.donotAutoReload = ['remixd'] // that would be a bad practice to force loading some plugins at page load.
```
Comment (GrandSchtroumpf) [inline src/remixAppManager.js]: I'm not sure to understand what it does.

File: src/app/files/remixd-handle.js (hunk 1 above)
Comment (yann300) [reply src/app/files/remixd-handle.js]: yep, there's already a link in the content of the modal

File: src/app/files/remixd-handle.js (hunk 2 above)
Comment (yann300) [reply src/app/files/remixd-handle.js]: updated

**Pull request merged**